src/library.js

```diff
@@ -2887,19 +2887,17 @@ addToLibrary({
 #endif
       // Floats are always passed as doubles, and doubles and int64s take up 8
       // bytes (two 32-bit slots) in memory, align reads to these:
-      buf += (ch != 105/*i*/) & buf;
+      buf += (ch != {{{ charCode('i') }}}) & buf;
+      readEmAsmArgsArray.push(
+        ch == {{{ charCode('i') }}} ? HEAP32[buf] :
 #if MEMORY64
-      // Special case for pointers under wasm64 which we read as int53 Numbers.
-      if (ch == 112/*p*/) {
-        readEmAsmArgsArray.push(readI53FromI64(buf++ << 2));
-      } else
+        // Special case for pointers under wasm64 which we read as int53 Numbers.
+        ch == {{{ charCode('p') }}} ? {{{ makeGetValue('buf++ << 2', 0, '*') }}} :
```

Comment:
This is a `? :` nested in another - is there no need for `( )` to avoid ambiguity in parsing..?

Reply:
Seems like the nesting works as we expect it to here.

```
$ cat test.js 
function f(n) {
  return n == 1 ? 'one' :
         n == 2 ? 'two' :
         n == 3 ? 'three' : NaN;
}

console.log(1);
console.log(2);
console.log(3);
$ node test.js
1
2
3
```

Reply:
Thanks for checking, might just be confusing for me personally then... :smile:  but seems fine in compiler output if it's unambiguous.